src/Arcus.Messaging.Pumps.ServiceBus/MessageHandling/AzureServiceBusFallbackMessageHandler.cs

```diff
@@ -0,0 +1,107 @@
+        /// <param name="message">The message that has to be dead lettered.</param>
+        /// <param name="newMessageProperties">The properties to modify on the message during the dead lettering of the message.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="message"/> is <c>null</c>.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the message handler was not initialized yet.</exception>
+        protected async Task DeadLetterAsync(Message message, IDictionary<string, object> newMessageProperties = null)
+        {
+            Guard.NotNull(message, nameof(message), "Requires a message to be dead lettered");
+
+            if (MessageReceiver is null)
+            {
+                throw new InvalidOperationException($"Cannot DeadLetter the message '{message.MessageId}' because the message receiver was not yet initialized yet");
+            }
+
+            Logger.LogTrace("DeadLettering message '{MessageId}'...", message.MessageId);
+            await MessageReceiver.DeadLetterAsync(message.SystemProperties.LockToken, newMessageProperties);
+            Logger.LogInformation("Message '{MessageId}' was DeadLettered!", message.MessageId);
+        }
+
+        /// <summary>
+        /// Dead letters the Azure Service Bus <paramref name="message"/> on Azure with a reason why the message needs to be dead lettered.
+        /// </summary>
+        /// <param name="message">The message that has to be dead lettered.</param>
+        /// <param name="deadLetterReason">The reason why the <paramref name="message"/> should be dead lettered.</param>
+        /// <param name="deadLetterErrorDescription">The optional extra description of the dead letter error.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="message"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="deadLetterReason"/> is blank.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the message handler was not initialized yet.</exception>
+        protected async Task DeadLetterAsync(Message message, string deadLetterReason, string deadLetterErrorDescription = null)
+        {
+            Guard.NotNull(message, nameof(message), "Requires a message to be dead lettered");
+            Guard.NotNullOrWhitespace(deadLetterReason, nameof(deadLetterReason), "Requires a non-blank dead letter reason for the message");
+
+            if (MessageReceiver is null)
+            {
+                throw new InvalidOperationException($"Cannot DeadLetter the message '{message.MessageId}' because the message receiver was not yet initialized yet");
+            }
+
+            Logger.LogTrace("DeadLettering message '{MessageId}'...", message.MessageId);
+            await MessageReceiver.DeadLetterAsync(message.SystemProperties.LockToken, deadLetterReason, deadLetterErrorDescription);
```

Comment:
Is there an overload that takes the 2 strings and the dictionary?

Reply:
Yeah, it takes in the `lockToken`, the `deadLetterReason` and an optional `deadLetterDescription`. 
https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.servicebus.core.messagereceiver.deadletterasync?view=azure-dotnet#Microsoft_Azure_ServiceBus_Core_MessageReceiver_DeadLetterAsync_System_String_System_String_System_String_

It seems logic that we provide all the overloads in these base classes, right?

Reply:
I was hoping it provided an overload with both but that's ok!